aip/general/0162.md

```diff
@@ -27,199 +27,174 @@ to the version of an API as a whole.
+### Creating revisions
 
-**Important:** APIs **must not** require a revision ID, and **must** default to
-the current revision if one is not provided, except in methods specifically
-dealing with the revision history (such as rollback) where failing to require
-it would not make sense (or lead to dangerous mistakes).
+Depending on the resource, different APIs may have different strategies for
+when to create a new revision, such as:
 
-### Getting a revision
+- Create a new revision any time that there is a change
+- Create a new revision when something important happens
+- Create a new revision when the user specifically asks
 
-APIs implementing resource revisions **should** accept a resource name with a
-revision ID in the standard `Get` method ([AIP-131][]):
+APIs **may** use any of these strategies. APIs that want to create a revision on
+user request should handle this with a custom method on the resource:
 
 ```proto
-message GetBookRequest {
-  // The name of the book.
-  //   Example: publishers/123/books/les-miserables
-  //
-  // In order to retrieve a previous revision of the book, also provide
-  // the revision ID.
-  //   Example: publishers/123/books/les-miserables@c7cfa2a8
+rpc CommitBookRevision(CommitBookRevisionRequest) returns (BookRevision) {
+  option (google.api.http) = {
+    post: "/v1/{name=publishers/*/books/*}:commitRevision"
+    body: "*"
+  };
+}
+
+message CommitBookRequest {
   string name = 1 [
     (google.api.field_behavior) = REQUIRED,
     (google.api.resource_reference) = {
       type: "library.googleapis.com/Book"
     }];
+  string tag = 2 [(google.api.field_behavior) = OPTIONAL];
 }
 ```
 
-If the user passes a revision ID that does not exist, the API **must** fail
-with a `NOT_FOUND` error.
+- The method **must** use the `POST` HTTP verb.
+- The method **should** return a long running operation or resource revision.
+- The request message **must** include the `name` field.
+- The resource ID of the resource revision **should** be a string and contain a
+  short, automatically-generated random string. A good rule of thumb is the last
+  eight characters of a [UUID version 4 (random)](UUID4).
+- `tag` field is optional. If provided, revision created will have the tag set.
+
+### Getting a revision
 
-APIs **must** return a `name` value corresponding to what the user sent. If the
-user sent a name with no revision ID, the returned `name` string **must not**
-include the revision ID either. Similarly, if the user sent a name with a
-revision ID, the returned `name` string **must** explicitly include it.
+Services `must` implement the standard `Get` method ([AIP-131][])
 
 ### Tagging revisions
 
+There are two types of tags: server and user specified.
+
+#### Server specified
+
+Services **may** reserve specific tags to be server-specified (e.g. `current`,
+`latest`), which are read-only and managed by the service.
+
+- If a `latest` tag exists, it **must** represents the most recent created
```

Comment:
```suggestion
- If a `latest` tag exists, it **must** represent the most recent created
```

Reply:
most recently created revision or the most recent version of the resource? This terminology might make it a little ambiguous.